File: lib/handlers/today-command-handler.js
'use strict';

const BaseCommandHandler = require('./base-command-handler');
const moment = require('moment');
const Menu = require('../../data/models/menu');

class TodayCommandHandler extends BaseCommandHandler {
  handle() {
    return new Promise((resolve, reject) => {
      Menu.findOne({date: moment().format('YYYYMMDD')}, (error, result) => {
        if (error)
          resolve('Error getting menu data.');
        else {
          if(!result)
          {
            resolve('I don\'t know which meals are being served today.!');
            return;
          }
          let message = `Today is the *${moment().format('DD.MM.YYYY')}* and the meals are:\n`;
          result.meals.forEach(meal => {
            message += `  ${meal}\n`;
          });
          resolve(message);
        }
      });
    });
  }
}

module.exports = TodayCommandHandler;

File: src/Deathstar.Canteen/Services/TodayCommandHandler.cs
using System;
using System.Threading;
using System.Threading.Tasks;
using Deathstar.Canteen.Persistence;

namespace Deathstar.Canteen.Services
{
	public class TodayCommandHandler : ICommandHandler
	{
		private readonly IMenuRepository menuRepository;
		private readonly ISlackbot slackbot;

		public TodayCommandHandler(IMenuRepository menuRepository, ISlackbot slackbot)
		{
			this.menuRepository = menuRepository;
			this.slackbot = slackbot;
		}

		public string HelpText { get; } = "The *today* command will return a list of today's meals.";

		public string SupportedCommandName { get; } = "today";

		public async Task HandleAsync(string arguments, string channel, CancellationToken cancellationToken)
		{
			DateTime date = DateTime.Today;
			Menu menu = await menuRepository.SingleOrDefaultAsync(x => x.Date == date.ToString("yyyyMMdd"), cancellationToken);
			string response = menu == null
				? "I don't know which meals are being served today!"
				: $"Today is the *{date:dd.MM.yyyy}* and the meals are:{Environment.NewLine}{menu}";

			slackbot.SendMessage(channel, response);
		}
	}
}
